src/main/java/com/esri/arcgisruntime/toolkit/skins/FeatureTemplatePickerTilePaneSkin.java

```diff
@@ -0,0 +1,329 @@
+    ToggleButton toggleButton = new ToggleButton(featureTemplateItem.getFeatureTemplate().getName());
+    toggleButton.setTooltip(new Tooltip(featureTemplateItem.getFeatureTemplate().getName()));
+    // add a style class so the item can be styled via CSS
+    toggleButton.getStyleClass().add("feature-template-item");
+    // add the item to a shared toggle group so only one feature template item can be selected at a time
+    toggleButton.setUserData(featureTemplateItem);
+    toggleButton.setToggleGroup(toggleGroup);
+    // bind the preferred width of the button based on the symbol size to limit overflowing template names
+    toggleButton.prefWidthProperty().bind(getSkinnable().symbolSizeProperty().multiply(2));
+    // use an image view for the button's graphic (to show the symbol swatch)
+    ImageView imageView = new ImageView();
+    toggleButton.setGraphic(imageView);
+    // update the swatch shown in the image view when the button is first created
+    updateSwatch(featureTemplateItem, imageView);
+    return toggleButton;
+  }
+
+  /**
+   * Updates the given image view with an image of the symbol swatch representing the feature template of the given
+   * feature template item.
+   *
+   * @param featureTemplateItem feature template item
+   * @param imageView image view to update
+   *
+   * @return completable future which completes when the image has been updated
+   */
+  private CompletableFuture<Void> updateSwatch(FeatureTemplateItem featureTemplateItem, ImageView imageView) {
+    // create a completable future to return
+    CompletableFuture<Void> updateFuture = new CompletableFuture<>();
+
+    // create a graphic based on the feature template item's feature template
+    final Graphic graphic = new Graphic();
+    graphic.getAttributes().putAll(featureTemplateItem.getFeatureTemplate().getPrototypeAttributes());
+    // get a symbol from the graphic based on the feature template item's feature layer's renderer
+    final FeatureLayer featureLayer = featureTemplateItem.getFeatureLayer();
+    Symbol symbol = featureLayer.getRenderer().getSymbol(graphic);
+    // create a swatch of the symbol according to the control's symbol size property
+    ListenableFuture<Image> swatch = symbol.createSwatchAsync(this.getSkinnable().getSymbolSize(),
+        this.getSkinnable().getSymbolSize(), (float) Screen.getPrimary().getOutputScaleX(), 0x00);
+    // update the image view's image with the swatch
+    swatch.addDoneListener(() -> {
+      try {
+        // ensure the image view's width and height match the symbol size property
+        imageView.setFitHeight(getSkinnable().getSymbolSize());
+        imageView.setFitWidth(getSkinnable().getSymbolSize());
+        imageView.setImage(swatch.get());
+        updateFuture.complete(null);
+      } catch (InterruptedException | ExecutionException | IllegalArgumentException e) {
+        updateFuture.completeExceptionally(e);
```

Comment:
Does the user see this exception somewhere?

Reply:
I've made an update to use `CompletableFuture.allOf(futures).whenCompleteAsync` which requests a new layout whether one of the futures completes exceptionally or not. So no, the exception will not bubble up to the user.